src/display.cpp

```diff
@@ -233,6 +233,8 @@ display::display(const display_context * dc, std::weak_ptr<wb::manager> wb, repo
 
 	set_idle_anim_rate(preferences::idle_anim_rate());
 
+	if(preferences::tile_size() > 0)
+		zoom_ = preferences::tile_size();
 	zoom_index_ = std::distance(zoom_levels.begin(), std::find(zoom_levels.begin(), zoom_levels.end(), zoom_));
```

Comment:
Here, `zoom_` is an `unsigned int` and elements of `zoom_levels` are doubles:

https://github.com/wesnoth/wesnoth/blob/b593e82cd268a380a8f14a1bb31debf1f1325477/src/display.hpp#L251-L255
https://github.com/wesnoth/wesnoth/blob/b593e82cd268a380a8f14a1bb31debf1f1325477/data/game_config.cfg#L35

Which of these would be more suitable to save in the preferences file?

Reply:
I stuck with `zoom_`/`tile_size` since that seems to be the figure that's actually used in the scaling calculations. Also storing integers as text seems to be cleaner than floating-point values (especially considering that one-third value). I didn't really consider zoom_levels when looking at how to implement this (again, not really sure what the 'right' way is since the relevant pieces of code seems to be scattered across several files).

Reply:
Agree about doubles. I just wonder if saving the `hex_size` by itself is sufficient, or whether we should save both `hex_size` and the reference value (72 pixels). After all, #1322 implies that it's _supposed_ to be possible to change the [game_config::tile_size](https://github.com/wesnoth/wesnoth/blob/9a4bd1342a41a24f7fdae6ba6babc5dee72d8486/src/game_config.cpp#L65-L73).